src/server/req_runjob.c

```diff
@@ -377,6 +381,9 @@ req_runjob(struct batch_request *preq)
 			/* job already running */
 			req_reject(PBSE_BADSTATE, 0, preq);
 			return;
+		} else if (get_subjob_discarding(parent, i) == 1) {
```

Comment:
Why are you passing in 'i' (state of the subjob). Shouldn't you be passing offset to this function? I'd also argue if the function is even needed.

Reply:
Yes, this should be offset, will change it.